onnxscript/rewriter/_rewrite_rule.py

```diff
@@ -157,6 +156,37 @@ def match(
                         _basics.MatchStatus.CONDITION_FAILED,
                     )
                 return None
+
+            try:
+                # Check node-level checkers
+                for pattern_node, ir_node in match.node_bindings.items():
+                    if pattern_node.check_method is not None:
+                        check_result = pattern_node.check_method(context, ir_node)
+                        if not check_result:
+                            return fail(
+                                check_result,
+                                f"Node-level check failed for pattern node {pattern_node}",
+                                ir_node,
+                            )
+
+                # Check value-level checkers
+                for pattern_value, ir_value in match.value_bindings.items():
+                    if pattern_value.check_method is not None:
+                        check_result = pattern_value.check_method(context, ir_value)
+                        if not check_result:
+                            return fail(
+                                check_result,
+                                f"Value-level check failed for pattern value {pattern_value}",
+                                ir_value,
+                            )
+
+                check_match_result = self._condition_function(context, **match.bindings)
+            except _basics.MatchFailureError as e:
```

Comment:
I want the try-except logic around each of the three checks, so that the failure messages and logic is specialized for each of the three. Do this by introducing a utility function `wrap_try(f)` that encapsulates the try-except pattern here. It should return a function that calls f within a try-catch block, catching MatchFailureError, and returning a MatchResult as below in the case of exceptions. Then call this in the 3 cases, like `wrap_try(self._condition_function)(context, **match.bindings)`